spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/observation/ObservationAutoConfiguration.java

```diff
@@ -67,21 +70,16 @@ ObservationRegistry observationRegistry() {
 	}
 
 	@Configuration(proxyBeanMethods = false)
-	@ConditionalOnBean(MeterRegistry.class)
```

Comment:
I thought that scenario not being supported is a "feature" since neither the grouping nor the `ObservationRegistryConfigurer` support it. I also think that this increases the scope of this PR, configuring `TracingAwareMeterObservationHandler` is just a subset of cases what this new requirement brings in.

I extended the PR with this: there are more auto-configuration now but maybe it is also easier to follow since all of the cases are covered not just the subset where we assumed that core is always there since it is a dependency of actuator.

Reply:
> core is always there since it is a dependency of actuator

`micrometer-core` is an optional dependency, as are all the other Micrometer modules.

There are really two separate things here. The first is that we need to ensure that things don't fail when they're not in an application's dependencies. This is the problem that I was raising with my comment above. The second is that we need to decide how things should behave when the different combinations are available. Thus far we have tried to support only metrics, only tracing, and metrics and tracing. There may be bugs in that attempt but I believe that's what we've been aiming for.